ReactSkia/components/RSkComponentImage.cpp

```diff
@@ -232,6 +219,103 @@ bool RSkComponentImage::processImageData(const char* path, char* response, int s
   return true;
 }
 
+inline void RSkComponentImage::drawContentShadow( SkCanvas *canvas,
+                            SkRect frameRect,
+                            SkRect targetRect,
+                            sk_sp<SkImage> imageData ,
+                            const ImageProps &imageProps,
+                            SkSize shadowOffset,
+                            SkColor shadowColor,
+                            float shadowOpacity){
+
+  SkRect shadowBounds;
+  SkIRect shadowFrame;
+  SkRect  frameBound;
+/*On below special cases, Shadow to be drawn on complete frame/Layout instead on Image target frame :
+  ---------------------------------------------------------------------------------------------------
+   1. The target size of Image > Frame's size. In that case, clipping will be done to contain the image
+      within the frame, So shadow to be drawn conidering frame size.
+   2. For Repeat mode, Target frame size will be handled by skia and it is not avialbel to teh components.
+*/
+  bool shadowOnFrame=(( frameRect.width() < targetRect.width()) || ( frameRect.height() < targetRect.height())||(imageProps.resizeMode == ImageResizeMode::Repeat));
+  if(shadowOnFrame) {
+    frameBound=frameRect;
+    shadowFrame.setXYWH(frameRect.x() + shadowOffset.width(), frameRect.y() + shadowOffset.height(), frameRect.width(), frameRect.height());
+  } else {
+    frameBound=targetRect;
+    shadowFrame.setXYWH(targetRect.x() + shadowOffset.width(), targetRect.y() + shadowOffset.height(), targetRect.width(), targetRect.height());
+  }
+  SkIRect shadowIBounds=RSkDrawUtils::getShadowBounds(shadowFrame,layer()->shadowMaskFilter,layer()->shadowImageFilter);
+  shadowBounds=SkRect::Make(shadowIBounds);
+
+  bool SaveLayerDone=false;
+//Apply Opacity
+  if(shadowOpacity) {
+    canvas->saveLayerAlpha(&shadowBounds,shadowOpacity);
+    SaveLayerDone=true;
+  }
+
+  SkPaint shadowPaint;
+  setImageFilters(shadowPaint,imageProps,targetRect,frameRect,true,imageData->isOpaque());
+
+  if(!imageData->isOpaque() ) {
+//Apply Shadow for transparent image
+    canvas->drawImageRect(imageData, targetRect, &shadowPaint);
+  } else {
+//Apply Shadow for opaque image
+    if(isOpaque(layer()->opacity)) {
+ // clipping done to avoid drawing on non visible area [Area under opque frame]
+      if(!SaveLayerDone) {
+        canvas->saveLayer(&shadowBounds,&shadowPaint);
+        SaveLayerDone =true;
+      }
+      canvas->clipRect(frameBound,SkClipOp::kDifference);
+    }
+    shadowPaint.setColor(shadowColor);
+    canvas->drawIRect(shadowFrame, shadowPaint);
+  }
+  if(SaveLayerDone) {
+    canvas->restore();
+  }
+#ifdef SHOW_SHADOW_BOUND
+  SkPaint paint;
+  paint.setStyle(SkPaint::kStroke_Style);
+  paint.setColor(SK_ColorGREEN);
+  paint.setStrokeWidth(2);
+  shadowBounds.join(frameBound);
+  canvas->drawRect(shadowBounds,paint);
+#endif
+}
+
+inline void RSkComponentImage::setImageFilters (SkPaint &paintObj,const ImageProps &imageProps,
```

Comment:
rename . setPaintFilters

And add comment ["This function apply appropriate filter on paint object to draw image & Shadow"]